torch/distributed/_shard/checkpoint/state_dict_loader.py

```diff
@@ -30,6 +34,27 @@
 
 from .api import CheckpointException
 
+def _create_shard_metadata(size: torch.Size, device: str) -> ShardMetadata:
+    rank = dist.get_rank() if dist.is_initialized() else 0
+    return ShardMetadata(
+        shard_offsets=[0] * len(size),
+        shard_sizes=list(size),
+        placement=f"rank:{rank}/{device}"
```

Comment:
It seems like we don't actually need the `placement` field here and need to fill it out just because `ShardMetadata` enforces this? If so, I'm wondering if we can relax this in ShardMetadata such that this can be `None` and in that case `ShardMetadata` is just referencing metadata of a ShardedTensor that hasn't really be placed anywhere yet physically (ex: checkpoint).

Reply:
Turns out ShardMetadata validation is quite centralized so this turned out to be pretty straightforward.

Reply:
I took the opportunity to do some minor changes to ShardMetadata and fix the typing of `placement`.